controller.py

```diff
@@ -380,19 +445,100 @@ def remove_items(self, collection_details):
 
         return feed_response(removal_feed)
 
-    def update_client_url(self):
-        """Updates the URL of a IntegrationClient"""
-        client = authenticated_client_from_request(self._db)
-        if isinstance(client, ProblemDetail):
-            return client
+    def register(self, do_get=HTTP.get_with_timeout):
+        opds_url = request.form.get('url')
```

Comment:
I recently changed the circulation/registry interface so that the URL submitted is the URL to the auth document itself (which must be publically available, no 401 allowed). Whether you keep the auth document in this role or replace it with a new type of document, I recommend making the corresponding change here. It eliminates a lot of special cases.